pyuvdata/uvdata.py

```diff
@@ -1407,10 +1407,18 @@ def _select_preprocess(self, antenna_nums, antenna_names, ant_str, bls,
 
         return blt_inds, freq_inds, pol_inds, history_update_string
 
-    def _select_metadata(self, blt_inds, freq_inds, pol_inds, history_update_string):
+    def _select_metadata(self, blt_inds, freq_inds, pol_inds, history_update_string,
+                         keep_missing_antennas=True):
         """
-        Internal function to perform select on everything except the data_array,
-        flag_array and nsample_array to allow for re-use in uvfits reading
+        Internal function to perform select on everything except the data-sized arrays.
+
+        Args:
+            blt_inds: list of baseline-time indices to keep. Can be None (to keep everything).
+            freq_inds: list of frequency indices to keep. Can be None (to keep everything).
+            pol_inds: list of polarization indices to keep. Can be None (to keep everything).
+            history_update_string: string to append to the end of the history.
+            keep_antennas: whether to keep metadata for antennas that are no longer in the dataset.
```

Comment:
The docstring keyword name should match the real keyword name. I think I'd slightly prefer something with metadata in the name, maybe `keep_all_metadata`?

Reply:
Good catch! I'll make sure it matches everywhere. And I like `keep_all_metdata`, I'll make that change too.